src/responder/nss/nsssrv_mmap_cache.c

```diff
@@ -548,17 +548,25 @@ static struct sss_mc_rec *sss_mc_find_record(struct sss_mc_ctx *mcc,
         }
 
         safealign_memcpy(&name_ptr, rec->data, sizeof(rel_ptr_t), NULL);
-        if (key->len > strs_len
```

Comment:
I do not think that dropping this check is good idea.

Have you consider a case when input `key->len` is really huge e.g. (8MiB)?

Reply:
We can add a check that key->len is e.g. not larger than SSS_NAME_MAX. But a huge input key is no indication that the memory cache is corrupted. But I guess this kind of check out be better in sss_mc_get_record(). Do you agree? Do you think SSS_NAME_MAX is a good limit or is there a better one?

Reply:
sssd-client can guarantee that input will not be longer then `SSS_NAME_MAX` which is usually 256. But I malicious user could use longer string pass directly via unix pipes therefore I mentioned 8MiB

For me `key->len > strs_len` means that input string cannot be in current record and this check can be used as shortcut(performance optimisation) for checking another entry in chain (with the same hash).

I think that such check is not wrong just action in true branch was wrong.

Reply:
We can have additional check before the current IF like this:
```
if (key->len > strs_len) {
    /* The string cannot be in current record */
    slot = sss_mc_next_slot_with_hash(rec, hash);
    continue;
}
```

I do not think such optimisation is needed because it will almost always be evaluated as false and in case it is evaluated as true it will only save us from doing few more checks. But maybe having this check makes this part of code more readable (at least it looks that way to me)? So I am OK with it.